vm/src/compiler.rs

```diff
@@ -226,38 +213,40 @@ impl CompilerEnv for TypeInfos {
     }
 }
 
-pub struct Compiler<'a> {
-    globals: &'a (CompilerEnv + 'a),
+pub struct Compiler<'a, E> {
```

Comment:
Eh, let's keep the weird &T impls for now. We should really do some profiling anyway before optimizing for stuff we don't need to. It would be nice to have cut out that extra code though...